packages/mobile/src/app/saga.test.ts

```diff
@@ -279,4 +320,41 @@ describe('App saga', () => {
       ])
       .run()
   })
+
+  it('Handles fetching over the air translations', async () => {
+    const translations = { someLang: { someNamespace: { someKey: 'someValue ' } } }
+    const timestamp = 123456
+    const appVersion = '1.0.0'
+    const mockedVersion = DeviceInfo.getVersion as jest.MockedFunction<typeof DeviceInfo.getVersion>
+    mockedVersion.mockImplementation(() => appVersion)
+    const defaultProviders: (EffectProviders | StaticProvider)[] = [
+      [select(allowOtaTranslationsSelector), true],
+      [select(otaTranslationsAppVersionSelector), appVersion],
+      [select(otaTranslationsLanguageSelector), 'en-US'],
+      [select(currentLanguageSelector), 'en-US'],
+      [select(otaTranslationsLastUpdateSelector), timestamp],
+      [call(saveOtaTranslations, { 'en-US': translations }), null],
+    ]
+
+    await expectSaga(handleFetchOtaTranslations)
+      .provide([[select(otaTranslationsLastUpdateSelector), 0], ...defaultProviders])
+      .put(setOtaTranslationsLastUpdate(timestamp, appVersion, 'en-US'))
+      .run()
+
+    await expectSaga(handleFetchOtaTranslations)
+      .provide([
+        [select(currentLanguageSelector), 'de'],
+        [call(saveOtaTranslations, { de: translations }), null],
+        ...defaultProviders,
+      ])
+      .put(setOtaTranslationsLastUpdate(timestamp, appVersion, 'de'))
+      .run()
+
+    await expectSaga(handleFetchOtaTranslations)
+      .provide([[select(otaTranslationsAppVersionSelector), '0.9.0'], ...defaultProviders])
+      .put(setOtaTranslationsLastUpdate(timestamp, appVersion, 'en-US'))
+      .run()
```

Comment:
Could you add comments explaining what each invocation of `handleFetchOtaTranslations` tests here so it's easier for people looking at it?

Reply:
updated!